### ISSUE TITLE
Postgres - Parsing failure for identifiers starting with _

### ISSUE THREAD

**Issue by pantierra** [sqlfluff/sqlfluff#1742]
Thanks for this great linter and actively developing and maintaining it!

## Expected Behaviour

Linter should run through and not fail.

## Observed Behaviour

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR REPLACE FUNCTION public.foo(_a...'
```

## Steps to Reproduce

```sql
CREATE OR REPLACE FUNCTION public.foo(_a TEXT, _b INT)
RETURNS FLOAT AS
$$
  RETURN 0.0
$$ LANGUAGE plpgsql STABLE PARALLEL SAFE;

```

## Dialect

PostgreSQL

## Version

* sqlfluff, version 0.7.1
* Python 3.9.7
